web-app/js/portal/ui/openlayers/layer/NcWMS.js

```diff
@@ -102,9 +102,7 @@ OpenLayers.Layer.NcWMS = OpenLayers.Class(OpenLayers.Layer.WMS, {
         });
     },
 
-    _timeSeriesDatesLoaded: function(datesWithData) {
```

Comment:
I actually think the function names make perfect sense. It is just a bit more complicated than the average filter as things are being done asynchronously + processing is mostly asynchronous. I had a second look at the function names and I could easily tell what each does. We can rename however `timeSeries` to just `times`, that can work.

I think we can do many things here. Generally speaking I'd like to see all of that factored into a normal filter, rather than this bastardization of NcWMS panel etc.

That, and the package name which I don't think is appropriate anymore.

On the bright side of things - it's stateless and relatively "bug free" so far.